nshlib/nsh_command.c

```diff
@@ -481,6 +481,10 @@ static const struct cmdmap_s g_cmdmap[] =
 #if CONFIG_NFILE_STREAMS > 0 && !defined(CONFIG_NSH_DISABLESCRIPT)
 #if !defined(CONFIG_NSH_DISABLE_SH) && defined(CONFIG_SYSTEM_SYSTEM)
   { "sh",       cmd_sh,       2, 2, "<script-path>" },
+#  ifdef CONFIG_SYSTEM_NSH_PROGNAME
+  { CONFIG_SYSTEM_NSH_PROGNAME,
+                cmd_sh,       1, 2, "<script-path>" },
```

Comment:
Will this break existing scripts? If so Please add a porting guide to this PR and note ALL the breaking changes that this PR is introducing.

Reply:
Yes, this change will forbid run nsh without argument. It's very dangerous because nsh:main will do some one time initialization by checking whether the argument is empty.